### DISCUSSION TITLE
Function.AutoPath — Type instantiation is excessively deep and possibly infinite.ts(2589)

### DISCUSSION THREAD

**Discussion by mesqueeb** [millsp/ts-toolbelt#217] [Q&A] [answered]
Thanks for [your answer](https://github.com/millsp/ts-toolbelt/issues/154#issuecomment-786775208) the other day!

Currently I have this issue:
> Type instantiation is excessively deep and possibly infinite.ts(2589)

When using this signature:
```ts
function pick<T extends Record<string, any>, K extends string> (obj: T, keys: F.AutoPath<T, K>[]): O.P.Pick<T, S.Split<K, '.'>>
```

It works like a charm if I try it with a payload like so:
```ts
const doc = { name: 'n1', id: '1', filled: true, notpicked: false }
const res = pick(doc, ['name', 'filled', 'id'])
```

however, it gives me that `infinite` error when the keys array is empty:
```ts
const res = pick(doc, [])
```

**Comment by millsp** [comment, marked as answer]
At the moment `AutoPath` only takes a single string:

```ts
function pick<T extends Record<string, any>, K extends string> (obj: T, path: F.AutoPath<T, K>): O.P.Pick<T, S.Split<K, '.'>>
```
